apps/clamav.sh

```diff
@@ -0,0 +1,142 @@
+then
+    # Ask for removal or reinstallation
+    reinstall_remove_menu
+    # Removal
+    apt purge clamav-daemon -y
+    apt purge clamav-freshclam -y
+    apt purge clamav -y
+    apt autoremove -y
+    rm -f /etc/systemd/system/clamav-daemon.service
+    rm -f "$SCRIPTS"/av_report.sh
+    crontab -u root -l | grep -v 'clamav-fullscan.sh'  | crontab -u root -
+    if is_app_installed files_antivirus
+    then
+        occ_command_no_check app:remove files_antivirus
+    fi
+    # Show successful uninstall if applicable
+    removal_popup
+else
+    print_text_in_color "$ICyan" "Installing ClamAV..."
+fi
+
+# Install needed tools
+apt update -q4 & spinner_loading
+apt install clamav clamav-freshclam clamav-daemon -y
+
+# stop freshclam and update the database
+check_command systemctl stop clamav-freshclam
+check_command freshclam
+check_command systemctl start clamav-freshclam
+
+# Edit ClamAV settings to fit the installation
+sed -i "s|^MaxDirectoryRecursion.*|MaxDirectoryRecursion 30|" /etc/clamav/clamd.conf
+sed -i "s|^MaxFileSize.*|MaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^PCREMaxFileSize.*|PCREMaxFileSize 100M|" /etc/clamav/clamd.conf
+sed -i "s|^StreamMaxLength.*|StreamMaxLength 100M|" /etc/clamav/clamd.conf
+
+# Start ClamAV
+check_command systemctl restart clamav-freshclam
+check_command systemctl restart clamav-daemon
+
+# Make the service more reliable
+check_command cp /lib/systemd/system/clamav-daemon.service /etc/systemd/system/clamav-daemon.service
+sed -i '/\[Service\]/a Restart=always' /etc/systemd/system/clamav-daemon.service
+sed -i '/\[Service\]/a RestartSec=3' /etc/systemd/system/clamav-daemon.service
+check_command systemctl daemon-reload
+check_command systemctl restart clamav-daemon
+
+# Install Nextcloud app
+install_and_enable_app files_antivirus
+
+# Configure Nextcloud app
+occ_command config:app:set files_antivirus av_mode --value="socket"
+occ_command config:app:set files_antivirus av_socket --value="/var/run/clamav/clamd.ctl"
+occ_command config:app:set files_antivirus av_stream_max_length --value="104857600"
+occ_command config:app:set files_antivirus av_max_file_size --value="-1"
+occ_command config:app:set files_antivirus av_infected_action --value="only_log"
+
+# Inform the user
+msg_box "ClamAV was succesfully installed.
+
+Your Nextcloud should be safer now."
+
+# Ask for full-scan
+if ! yesno_box_yes "Do you want to setup a weekly full-scan of all files?
+It will run on sundays starting at 10:00 for max 12h. 
+You will be notified what the result was."
+then
+    exit
+fi
+
+# Create the full-scan script
+cat << CLAMAV_REPORT > "$SCRIPTS"/clamav-fullscan.sh
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+AV_REPORT="$(clamscan \
+--recursive \
+--stdout \
+--infected \
+--cross-fs \
+--log=$VMLOGS/clamav-fullscan.log \
+--max-scantime=43200000 \
+--max-filesize=100M \
```

Comment:
Does 12h max-scantime makes sense? (the value is in ms)
Also, does 100 MB make sense?

Reply:
We could also add the option to move infected files to a different directory...

Reply:
Problem is this is written for max filesize:
`Warning: disabling this limit or setting it too high may result in severe damage to the system.`

Reply:
https://manpages.debian.org/testing/clamav-daemon/clamd.conf.5.en.html